### PR TITLE
Code Insights: Connect drill-down UI with BE insight loading card logic

### PR DESCRIPTION
**Pull request by vovakulikov**
### Context
Closes https://github.com/sourcegraph/sourcegraph/issues/23190

This PR connects the drill-down filters panel component and BE fetch logic with applied drill-down filters. 
We already have drill-down filters in our gql API. 

```gql
points(
   from: DateTime
   to: DateTime
   includeRepoRegex: String
   excludeRepoRegex: String
): [InsightDataPoint!]! {
   dateTime,
   value
}
```

Data points over a time range (inclusive)
- If no 'from' time range is specified, the last 90 days of data is assumed.
- If no 'to' time range is specified, the current point in time is assumed.
- **includeRepoRegex** will include in the aggregation any repository names that match the provided regex
- **excludeRepoRegex** will exclude in the aggregation any repository names that match the provided regex

And we already have a drill-down filter panel component from this PR https://github.com/sourcegraph/sourcegraph/pull/23559

### How to test
You must have BE insights in your dashboard (to create this type of insight you have to go to insight creation UI and turn on Run over all repos checkbox - one import thing currently this checkbox and this mode of BE insights visible only for users that have `codeInsightsAllRepos: true` in they `experimentalFeatures` settings

Right after that, you must see your BE insights on the dashboard page with a filter icon right next to the three dots menu. By this filter icon you can set your drill-down filters.

### What have been done
- [x] Connect the filter panel component with the top-level state at BE insight card
- [x] Apply filters value in the BE fetch method (gql API handler)

### CODE REVIEW

File: client/web/src/insights/components/insights-view-grid/components/backend-insight/BackendInsight.story.tsx [hunk 1]
```diff
@@ -28,21 +29,22 @@ const INSIGHT_CONFIGURATION_MOCK: SearchBackendBasedInsight = {
 }
 
 const mockInsightAPI = createMockInsightAPI({
-    getBackendInsightById: (id: string) =>
-        of({
+    getBackendInsightById: (id: string, filters) => {
+        console.log('RUN API WITH', filters)
```
Comment (tjkandala) [inline client/web/src/insights/components/insights-view-grid/components/backend-insight/BackendInsight.story.tsx:33]:
```suggestion
```

File: client/web/src/insights/components/insights-view-grid/components/backend-insight/BackendInsight.story.tsx [hunk 2]
```diff
@@ -28,21 +29,22 @@ const INSIGHT_CONFIGURATION_MOCK: SearchBackendBasedInsight = {
...
+        return of({
             id,
             view: {
                 title: 'Backend Insight Mock',
                 subtitle: 'Backend insight description text',
                 content: [LINE_CHART_CONTENT_MOCK],
             },
-        }),
+        }).pipe(delay(2000))
```
Comment (tjkandala) [inline client/web/src/insights/components/insights-view-grid/components/backend-insight/BackendInsight.story.tsx:41]: Is this delay intentional? Our [storybook test](https://www.chromatic.com/pullrequest?activeElementId=comparison-test-key-610945a4a36808f345011cf9-1200&appId=5f0f381c0e50750022dc6bf7&number=23623&view=changes) doesn't test the insight anymore, only the loading state

File: client/web/src/insights/components/insights-view-grid/components/backend-insight/components/drill-down-filters-panel/DrillDownFiltersPanel.tsx [hunk 3]
```diff
@@ -4,62 +4,98 @@ import React, { forwardRef, InputHTMLAttributes, PropsWithChildren, Ref } from '
...
 
-    const { ref } = useForm<DrillDownFilters>({
-        initialValues: filters,
-        onChange: values => console.log(values),
+    const mode = useField({
+        name: 'mode',
+        formApi: formAPI,
     })
 
+    const includeRegex = useField({
```
Comment (tjkandala) [inline client/web/src/insights/components/insights-view-grid/components/backend-insight/components/drill-down-filters-panel/DrillDownFiltersPanel.tsx:33]: nit unrelated to this PR: I think `useFieldAPI` should be `UseFieldAPI` (interfaces are usually in PascalCase)

File: client/web/src/insights/hooks/use-parallel-requests/use-parallel-request.ts [hunk 4]
```diff
@@ -101,6 +101,10 @@ export function createUseParallelRequestsHook<T>({ maxRequests } = { maxRequests
         useEffect(() => {
             const cancelStream = new Subject<boolean>()
 
+            console.log('Start fetch use parallel')
```
Comment (tjkandala) [inline client/web/src/insights/hooks/use-parallel-requests/use-parallel-request.ts:104]:
```suggestion
```

**Review by tjkandala — APPROVED** [review]
LGTM! I didn't get to manually test it since I'm out of office, but I'm excited to try this feature out

File: client/web/src/insights/components/insights-view-grid/components/backend-insight/BackendInsight.story.tsx (hunk 2 above)
Comment (vovakulikov) [reply client/web/src/insights/components/insights-view-grid/components/backend-insight/BackendInsight.story.tsx:41]: Nice catch. You're about loading state. I added this cause a want to test insight dill-down denounce filter change logic. But probably I should add another story for that.

File: client/web/src/insights/components/insights-view-grid/components/backend-insight/components/drill-down-filters-panel/DrillDownFiltersPanel.tsx (hunk 3 above)
Comment (vovakulikov) [reply client/web/src/insights/components/insights-view-grid/components/backend-insight/components/drill-down-filters-panel/DrillDownFiltersPanel.tsx:33]: Agree. I will rename this in a separate PR, Thanks

**Comment by vovakulikov** [comment]
**For reviewers**
At the moment, there may be technical difficulties to test these filters manually since we have problems with BE insights on k8s and sg.com stagings.

**Pull request merged**